### PR TITLE
Add Environmental Stewardship

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: app/processing/genesis/filter-payment-requests.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const addErrorMessage = (paymentRequest, errorMessage) => {
+  if (paymentRequest.errorMessage) {
+    paymentRequest.errorMessage += errorMessage
+  } else {
+    paymentRequest.errorMessage = `Payment request for FRN: ${paymentRequest.frn} - ${paymentRequest.invoiceNumber} is invalid, ${errorMessage}`
+  }
+}
+
+module.exports = filterPaymentRequest
```
Comment (simonnineteen90) [inline app/processing/genesis/filter-payment-requests.js:62]: this file could probably beneift from being split out a bit more but its probably out of scope for this but could be worth adding a ticket to backlog to look at this file for and the similar ones in the repo

File: app/processing/genesis/transform-header.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const { v4: uuidv4 } = require('uuid')
+
+const transformHeader = (headerData, filename) => {
+  return {
+    correlationId: uuidv4(),
+    batch: filename,
+    invoiceNumber: headerData[1],
+    paymentRequestNumber: 1,
```
Comment (simonnineteen90) [inline app/processing/genesis/transform-header.js:8]: Is this correct to be hard-coded to `1`? Does this scheme only have first payments? Should it be set as a constant?

File: app/processing/genesis/transform-invoice-line.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const transformInvoiceLine = (lineData) => {
+  return [{
+    companyCode: lineData[2],
```
Comment (simonnineteen90) [inline app/processing/genesis/transform-invoice-line.js:3]: is it worth a comment or updating the docs somewhere to explain why there are two invoice lines here as it contrast with the other schemes?

File: test/unit/processing/genesis/transform-invoice-lines.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      companyCode: '',
+      costCentre: '',
+      description: '',
+      projectCode: undefined,
+      standardCode: '',
+      subAccountCode: '',
+      value: NaN
+    }])
+  })
+})
```
Comment (simonnineteen90) [inline test/unit/processing/genesis/transform-invoice-lines.test.js:27]: might need another test here to cover when there are values in the second half of the  invoice line

File: app/processing/validate-sequence.js [hunk 5]
```diff
@@ -19,6 +19,8 @@ const getSequence = (schemeId, filename) => {
     case bps.schemeId:
     case fdmr.schemeId:
       return Number(filename.substr(5, 4))
+    case es.schemeId:
+      return Number(filename.substr(24, 4))
```
Comment (simonnineteen90) [inline app/processing/validate-sequence.js:23]: should this be `Number(filename.substr(23, 4)`, think you are missing the first digit unless its not needed? Could add a unit test as well. 

I tested on this filename and missed first digit: `GENESISPayReq_20230104_5394.gne`

File: app/processing/genesis/filter-payment-requests.js (hunk 1 above)
Comment (johnwatson484) [reply app/processing/genesis/filter-payment-requests.js:62]: Yeah agreed, we've already got an improvement ticket to do that for this and also the Request Editor.

The names of some of the modules should also be considered as part of that too.

File: app/processing/genesis/transform-header.js (hunk 2 above)
Comment (johnwatson484) [reply app/processing/genesis/transform-header.js:8]: Yes, Genesis only produces first payments for ES.  I did consider setting it as a constant but given it's not used anywhere else and it's very clear what the property and value represent it seems a bit redundant to create files and references just for a number `1`.

File: app/processing/genesis/transform-invoice-line.js (hunk 3 above)
Comment (johnwatson484) [reply app/processing/genesis/transform-invoice-line.js:5]: Added a comment to explain the reasoning

File: test/unit/processing/genesis/transform-invoice-lines.test.js (hunk 4 above)
Comment (johnwatson484) [reply test/unit/processing/genesis/transform-invoice-lines.test.js:51]: Agreed, test added.

File: app/processing/validate-sequence.js (hunk 5 above)
Comment (johnwatson484) [reply app/processing/validate-sequence.js]: Ooops good spot.  Fixed and tests added.

**Review by simonnineteen90 — APPROVED** [review]

**Pull request merged**